app/services/auth.py

```diff
@@ -0,0 +1,100 @@
+from configparser import ConfigParser
+import os
+from typing import Optional
+
+import jwt
+from fastapi import Depends, HTTPException, status
+from fastapi.security import SecurityScopes, HTTPAuthorizationCredentials, HTTPBearer
+
+def get_auth0_config():
+    """Sets up configuration for the app"""
+
+    env = os.getenv("ENV", ".config")
+
+    # Read config from .config file or environment variables
+    if env == ".config":
+        config = ConfigParser()
+        config.read(".config")
+        config = config["AUTH0"]
+    else:
+        config = {
+            "DOMAIN": os.getenv("AUTH0_DOMAIN"),
+            "API_AUDIENCE": os.getenv("AUTH0_API_AUDIENCE"),
+            "ISSUER": os.getenv("AUTH0_ISSUER"),
+            "ALGORITHMS": os.getenv("AUTH0_ALGORITHMS"),
+        }
+    for key, value in config.items():
+        if value == "":
+            print(f"Missing config: {key}")
+            exit(1)
+    return config
+
+class UnauthorizedException(HTTPException):
+    def __init__(self, detail: str, **kwargs):
+        """Returns HTTP 403"""
+        super().__init__(status.HTTP_403_FORBIDDEN, detail=detail)
+
+
+class UnauthenticatedException(HTTPException):
+    def __init__(self):
+        super().__init__(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="Requires authentication"
+        )
+
+
+class VerifyToken:
+    """Does all the token verification using PyJWT"""
+
+    def __init__(self):
+        self.config = get_auth0_config()
+
+        # This gets the JWKS from a given URL and does processing so you can
+        # use any of the keys available
+        jwks_url = f'https://{self.config["DOMAIN"]}/.well-known/jwks.json'
```

Comment:
can you log somewhere which env variables we will need to configure in the prod environment when it is up so we wont need to debug it?

Reply:
It's in .config which is committed to the repo
Obviously not the best practice but I believe it's OK for now